depthai_helpers/config_manager.py

```diff
@@ -0,0 +1,342 @@
+import os
+import json
+import platform
+import subprocess
+from pathlib import Path
+import consts.resource_paths
+
+from depthai_helpers.model_downloader import download_model
+from depthai_helpers.cli_utils import cli_print, PrintColors
+
+class DepthConfigManager:
+    labels = ""
+
+    def __init__(self, args):
+        self.args = args
+        self.stream_list = args['streams']
+        self.compile_model = self.args['shaves'] is not None and self.args['cmx_slices'] is not None and self.args['NN_engines']
+        self.calc_dist_to_bb = not self.args['disable_depth']
+
+        # Prepare handler methods (decode_nn, show_nn) for the network we want to run.
+        self.decode_nn, self.show_nn, self.decode_nn_json = self.importAndGetCallbacksForNN()
```

Comment:
maybe it would be cleaner/easier to use `@property` annotation (for `self.jsonConfig` also)?

Reply:
Hmm, yeah, I guess returning that many variables isn't pretty. l can change these to set the class variables in the methods instead.

Reply:
For the `self.jsonConfig`, you could do

```python
@property
def jsonConfig(self):
    # body of postProcessJsonConfig method
```

and then when you access `self.jsonConfig`, you treat it as a regular variable, but it will actually run the function to resolve it's value.
But for the `decode_nn` and others - it's fine for me to leave it as it is, I just didn't like that we assign to `self` both here and inside `self.importAndGetCallbacksForNN()`